src/Message/ApplePayAuthorizeRequest.php

```diff
@@ -0,0 +1,362 @@
+<?php
+
+namespace Omnipay\Vindicia\Message;
+
+use Omnipay\Common\Message\ResponseInterface;
+
+/**
+ * Retrieve an Apple Pay Payment Session object.
+ *
+ * Post a request to Apple Pay server's Payment Session endpoint using two-way TLS.
+ *
+ * Parameters:
+ * - pemCertPath: Path to the cert of the split Merchant Identity certification,
+ * associated with your merchantID needed to make the call.
+ * - keyCertPath: Path to the key cert of the split Merchant Identity certification,
+ * associated with your merchantID needed to make the call.
+ * - keyCertPassword: Key certification password for the key certification.
+ * - validationUrl: Validation URL received from session.onvalidatemerchant on the client.
+ * - merchantIdentifier: Your merchant ID provided by Apple.
+ * - displayName: A name for your store, suitable for display, appears in the touch bar.
+ * - applicationType: The "initiative" parameter depending on the type of application.
+ * For Apple Pay on the web, use "web". For Business Chat, use "messaging".
+ * - applicationUrl: The initive context is based on the value supplied for the applicationType/initiative.
+ * For Apple Pay on the web, provide your fully qualified domain name associated with your
+ * Apple Pay Merchant Identity Certificate. For Business Chat, pass your payment gateway URL.
+ * See @link for more details.
+ *
+ * Example:
+ *
+ * <code>
+ *   // set up the gateway
+ *   $gateway = \Omnipay\Omnipay::create('Vindicia ApplePay');
+ *   $gateway->setUsername('your_username');
+ *   $gateway->setPassword('y0ur_p4ssw0rd');
+ *   $gateway->setPemCertPath('./certs/path_to_cert.crt.pem');
+ *   $gateway->setKeyCertPath('./certs/path_to_key_cert.key.pem');
+ *   $gateway->setKeyCerPassword('y0ur_key_p4ssw0rd');
+ *   $gateway->setTestMode(false);
+ *
+ *   // authorize the request and get an Apple Pay Session object.
+ *   $ApplePaySessionObject = $gateway->authorize(array(
```

Comment:
it returns a response, right? not an apple pay object?

Reply:
I will need to connect all the parts to verify how I would send it back. Essentially, what you get back is what you need for the Apple Pay Session.

Reply:
right, but the authorize function returns an Omnipay response. So there's something specific in the Omnipay response that actually gets passed to the front end. You'll need to figure out what that is in this PR so that you can make sure it is accessible on the response and named and documented appropriately.

Reply:
Well essentially the front end should get the entire data object. I'm thinking I could have a function to return all the of the data.

Reply:
Yeah if the front end needs an entire object that's returned by apple, than having a function to retrieve that object from the omnipay response makes sense

Reply:
could you update this to match the docs in the gateway file? the gateway file uses abstractResponse as the name here, which is more correct, since it returns a response, not an apple pay object

Reply:
this still needs to be addressed. i'd recommend just copying the example from the gateway to here because i think that one is all up to date

Reply:
Oh, I guess I didn't fully understand what you meant. It's the gateway code example that you want in the request docs? I thought to keep them separate since it's in the request.

Reply:
Well what I mean is just that you're still calling this $ApplePaySessionObject, not $authorizeResponse. Although you're calling it $authorizeResponse below. And you've got it right in the gateway request docs. So just fixing that is fine.

Reply:
Ohhh, okay, gotcha now.